examples/developer-tools/src/blocks/fields.ts

```diff
@@ -0,0 +1,395 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly';
+
+/**
+ * Check to see if more than one field has this name.
+ * Highly inefficient (On^2), but n is small.
+ * @param referenceBlock Block to check.
+ */
+const fieldNameCheck = function(referenceBlock: Blockly.Block) {
+  if (!referenceBlock.workspace) {
+    // Block has been deleted.
+    return;
+  }
+  const name = referenceBlock.getFieldValue('FIELDNAME').toLowerCase();
+  let count = 0;
+  const blocks = referenceBlock.workspace.getAllBlocks(false);
+  for (const block of blocks) {
+    const otherName = block.getFieldValue('FIELDNAME');
+    if (block.isEnabled() && !block.getInheritedDisabled() &&
+        otherName && otherName.toLowerCase() === name) {
+      count++;
+    }
+  }
+  const msg = (count > 1) ?
+      'There are ' + count + ' field blocks\n with this name.' : null;
+  referenceBlock.setWarningText(msg);
+};
+
+/**
+ * Label (non-serializable) field.
+ */
+export const fieldLabel = {
```

Comment:
This gets complicated if I'm trying to export the block definitions so that they can be registered from a different file than they're defined in. I would have to know whether the thing I'm exporting is a JS definition or a JSON definition, and then call them differently in the register call. I think it's simpler and less prone to error to define and export them all consistently even though that means using JS when I could use JSON.

We have been trying to remove side effects from our files but in core we still register blocks in the file we declare them so I don't have a good pattern to follow. I'm open to suggestions if this is important to you.